pkg/config/config_template.yaml

```diff
@@ -1,424 +1,781 @@
+## Metadata providers, add or remove from the list to enable or disable collection.
+## Intervals are expressed in seconds. You can also set a provider's interval to 0
+## to disable it.
+#
+#  metadata_providers:
+#    - name: k8s
+#      interval: 60
+
 {{ end -}}
 {{- if .JMX }}
-# JMX
+
+#############################
+## JMX Agent Configuration ##
+#############################
+
+## @param jmx_custom_jars - list of strings - optional
+## If you only run Autodiscovery tests, jmxfetch might fail to pick up custom_jar_paths
+## set in the check templates. If that is the case, force custom jars here.
+#
+#  jmx_custom_jars:
+#    - /jmx-jars/jboss-cli-client.jar
+
+## @param jmx_use_cgroup_memory_limit - boolean - optional - default: false
+## When running in a memory cgroup, openjdk 8u131 and higher can automatically adjust
+## its heap memory usage in accordance to the cgroup/container's memory limit.
+## The Agent set a Xmx of 200MB if none is configured.
+## Note: older openjdk versions and other jvms might fail to start if this option is set
+#
+#  jmx_use_cgroup_memory_limit: false
+
+## @param jmx_max_restarts - integer - optional - default: 3
+## Number of JMX restarts allowed in the restart-interval before giving up.
+#
+#  jmx_max_restarts: 3
+
+## @param jmx_restart_interval - integer - optional - default: 5
+## Duration of the restart interval in seconds.
+#
+#  jmx_restart_interval: 5
+
+## @param jmx_thread_pool_size - integer - optional - default: 3
+## JMXFetch collects multiples instances concurrently. The following options may
+## help fine-tune the level of concurrency and timeouts that come into play during the
+## collection of metrics from configured instances:
+## Defines the maximum level of concurrency:
+##   * Higher concurrency increases CPU utilization during metric collection.
+##   * Lower concurrency results in lower CPU usage but may increase the total collection time.
+## A value of 1 processes instances serially.
+#
+#  jmx_thread_pool_size: 3
+
+## @param jmx_collection_timeout - integer - optional - default: 60
+## Defines the maximum waiting period in seconds before timing up on metric collection.
+#
+#  jmx_collection_timeout: 60
+
+## @param jmx_reconnection_thread_pool_size - integer - optional - default: 3
+## Defines the maximum level of concurrency:
+##   * Higher concurrency increases CPU utilization during reconnection.
+##   * Lower concurrency results in lower CPU usage but may increase the total reconnection time
+## A value of 1 processes instance reconnections serially.
+#
+#  jmx_reconnection_thread_pool_size: 3
+
+## @param jmx_reconnection_timeout - integer - optional - default: 10
+## Determines the maximum waiting period in seconds before timing up on instance reconnection.
+#
+#  jmx_reconnection_timeout: 10
+
+{{ end -}}
+{{- if .Logging }}
+
+#################################
+## Agent Logging Configuration ##
+#################################
+
```

Comment:
Should we add some comments here that explain that this is made only to configure how the Datadog Agent logs and is not related to log collection?